### PR TITLE
Added authorization requirements for posting and deleting reports

### LINKED ISSUE #397: Add authentication/admin checks to report api routes
There is no authentication/admin checking on the reports API routes. The get and delete routes should only be accessible to an authenticated user who is an admin.

### PR DESCRIPTION
**Pull request by timobraz**
## Description

if passport doesn't exist or no admin value on passport then you fail to access GET or DELETE routes. Still works if you're authenticated and admin status.

## Screenshots
Screenshot 2024-01-31 at 6 00 51 PM

## Steps to verify/test this change:
- [ ] Verify changes work as expected on staging instance

## Final Checks:
- [ ] Verify successful deployment

(optional)
- [ ] Write tests
- [ ] Write documentation

## Issues

Closes #397

### CODE REVIEW

File: api/src/controllers/reports.ts [hunk 1]
```diff
@@ -13,6 +13,8 @@ const router = express.Router();
  * Get all reports
  */
 router.get('/', async (req, res) => {
+  if (!req.session.passport) return res.status(401).send('Unathenticated');
```
Comment (js0mmer) [inline api/src/controllers/reports.ts:16]:
```suggestion
  if (!req.session.passport) return res.status(401).send('Unauthenticated');
```

File: api/src/controllers/reports.ts [hunk 2]
```diff
@@ -34,6 +36,8 @@ router.post('/', async (req, res) => {
  */
 router.delete('/', async (req, res) => {
   let status;
+  if (!req.session.passport) return res.status(401).send('Unathenticated');
```
Comment (js0mmer) [inline api/src/controllers/reports.ts:39]:
```suggestion
  if (!req.session.passport) return res.status(401).send('Unauthenticated');
```

**Review by js0mmer — APPROVED** [review]
LGTM! Just left comments on a couple typos that don't really matter

File: api/src/controllers/reports.ts (hunk 1 above)
Comment (timobraz) [reply api/src/controllers/reports.ts:16]: lmfaooo that's my bad

**Pull request merged**